paddle/fluid/operators/activation_mkldnn_op.cc

```diff
@@ -115,11 +119,13 @@ void eltwise_forward(const framework::ExecutionContext &ctx,
   const std::string key_fwd = key_with_layout + "@eltwise_fwd";
   const std::string key_fwd_pd = key_with_layout + "@eltwise_fwd_pd";
 
+  bool is_test = ctx.Attr<bool>("is_test");
+
   // save input data and layout to be referred in backward path
   auto p_src_data = std::make_shared<const T *>(x_data);
-  dev_ctx.SetBlob(key_src_data, p_src_data);
+  if (!is_test) dev_ctx.SetBlob(key_src_data, p_src_data);
   auto p_src_layout = std::make_shared<memory::format>(src_format);
-  dev_ctx.SetBlob(key_src_layout, p_src_layout);
+  if (!is_test) dev_ctx.SetBlob(key_src_layout, p_src_layout);
```

Comment:
Are variables `p_src_data` and `p_src_layout` needed at inference? Are these shared pointers passed by value or by reference? You could potentially turn two if-statements in lines 126 and 128 into one, if these two shared pointers are passed by value to `SetBlob`.

Reply:
I didn't see any usage of them. I was actually hoping that a reviewer would verify if I didn't break things.
They are passed by value:
```
void MKLDNNDeviceContext::SetBlob(const std::string& name,           
                                  std::shared_ptr<void> data) const {
```
Ok I put both in one if().